dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,325 @@
+import math
+import numpy as np
+import toolz
+from sklearn.base import clone
+from sklearn.model_selection import ParameterSampler
+from sklearn.utils import check_random_state
+
+from tornado import gen
+from toolz import groupby
+from dask.distributed import as_completed, default_client, futures_of
+from dask_searchcv.model_selection import DaskBaseSearchCV
+import dask.array as da
+from ..datasets import make_classification
+from ..wrappers import Incremental
+from ._split import train_test_split
+from distributed.metrics import time
+from distributed import Future
+from copy import deepcopy
+
+
+def _get_nr(R, eta=3):
+    s_max = math.floor(math.log(R, eta))
+    B = (s_max + 1) * R
+
+    brackets = list(reversed(range(int(s_max + 1))))
+    N = [math.ceil(B / R * eta**s / (s + 1)) for s in brackets]
+    R = [int(R * eta**-s) for s in brackets]
+    return list(map(int, N)), list(map(int, R)), brackets
+
+
+def _partial_fit(model_and_meta, X, y, meta=None, fit_params={},
+                 dry_run=False, **kwargs):
+    start = time()
+    assert isinstance(X, np.ndarray)
+    assert y is None or isinstance(y, np.ndarray)
+    model = deepcopy(model_and_meta[0])
```

Comment:
I wonder if we might use the `safe=False` keyword to clone here

Reply:
This is still present in https://github.com/dask/dask-ml/blob/ce81ad12b156548ca31b447bf34ab1ba119e9469/dask_ml/model_selection/_incremental.py#L79

I'm only seeing `memo` and `_nil` parameters to `deepcopy`, and get an exception when I pass `safe=True` to `copy.deepcopy`.